src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java

```diff
@@ -0,0 +1,146 @@
+package com.mathworks.ci.freestyle;
+
+/**
+ * Copyright 2019-2024 The MathWorks, Inc.
+ * 
+ * Script builder used to run custom MATLAB commands or scripts.
+ * 
+ */
+
+import java.io.IOException;
+import javax.annotation.Nonnull;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.StaplerRequest;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.init.Initializer;
+import hudson.init.InitMilestone;
+import hudson.model.Items;
+import hudson.model.AbstractProject;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import jenkins.tasks.SimpleBuildStep;
+import net.sf.json.JSONObject;
+
+import com.mathworks.ci.Message;
+import com.mathworks.ci.parameters.RunActionParameters;
+import com.mathworks.ci.actions.MatlabActionFactory;
+import com.mathworks.ci.actions.RunMatlabCommandAction;
+import com.mathworks.ci.freestyle.options.StartupOptions;
+
+public class RunMatlabCommandBuilder extends Builder implements SimpleBuildStep {
+    // Deprecated
+    private transient int buildResult;

```

Comment:
Unfortunately removing a given field rather than just setting it to transient can sometimes mess up the deserialization of past builds. For backwards compatibility the recommended approach is to instead switch the field to transient so it will no longer be serialized in future jobs but is still compatible with previous jobs.